Deferred/FutureCollections.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  FutureCollections.swift
+//  Deferred
+//
+//  Created by Zachary Waldowski on 8/29/15.
+//  Copyright © 2014-2015 Big Nerd Ranch. Licensed under MIT.
+//
+
+import Dispatch
+
+public extension SequenceType where Generator.Element: FutureType {
+    /// Choose the future that is determined first from a collection of futures.
+    ///
+    /// - returns: A deferred value that is determined with the first of the
+    ///   given futures to be determined.
+    var firstFuture: Deferred<Generator.Element.Value> {
+        let combined = Deferred<Generator.Element.Value>()
+        for d in self {
+            d.upon { t in combined.fill(t, assertIfFilled: false) }
+        }
+        return combined
+    }
+}
+
+public extension CollectionType where Generator.Element: FutureType {
+    /// Compose a number of futures into a single deferred array.
+    ///
+    /// - returns: A deferred array that is determined once all the given values
+    ///   are determined, in the same order.
+    var allFutures: Deferred<[Generator.Element.Value]> {
```

Comment:
`determinedValues`? `joinPoint` / `joinedValue` (harks back to the fork/join description I read in the docstring on FutureType IIRC)? `allTogether`?